config/config.go

```diff
@@ -329,10 +341,24 @@ func (conf *TerraformConfig) GetAfterHooks() []Hook {
 	return conf.AfterHooks
 }
 
+func (conf *TerraformConfig) GetErrorHooks() []ErrorHook {
+	if conf == nil {
+		return nil
+	}
+
+	return conf.ErrorHooks
+}
+
 func (conf *TerraformConfig) ValidateHooks() error {
-	allHooks := append(conf.GetBeforeHooks(), conf.GetAfterHooks()...)
+	beforeAndAfterHooks := append(conf.GetBeforeHooks(), conf.GetAfterHooks()...)
+
+	for _, curHook := range beforeAndAfterHooks {
+		if len(curHook.Execute) < 1 || curHook.Execute[0] == "" {
+			return InvalidArgError(fmt.Sprintf("Error with hook %s. Need at least one non-empty argument in 'execute'.", curHook.Name))
+		}
+	}
 
-	for _, curHook := range allHooks {
+	for _, curHook := range conf.GetErrorHooks() {
```

Comment:
Hmm.. so we run the before and after hooks first, then we run the error hooks. We should mention this somewhere in the docs.

Reply:
This is addressed https://github.com/gruntwork-io/terragrunt/pull/1967/commits/94037c72168be46d1cb7dbb47d687aea0933148c